src/GitVersionExe/Program.cs

```diff
@@ -129,17 +129,27 @@ static void ConfigureLogging(Arguments arguments)
             {
                 try
                 {
-                    Directory.CreateDirectory(Path.GetDirectoryName(arguments.LogFilePath));
-                    if (File.Exists(arguments.LogFilePath))
+                    var logFileFullPath = Path.GetFullPath(arguments.LogFilePath);
+                    var logFile = new FileInfo(logFileFullPath);
+
+                    if (logFile.Directory != null && !logFile.Directory.Exists)
```

Comment:
We should probably throw if `logFile.Directory == null` and then just call `logFile.Directory.Create();` no need for recursion in this case.